src/modules/launcher/Wox.Plugin/ThemeManager.cs

```diff
@@ -90,67 +89,73 @@ private static Theme GetHighContrastBaseType()
         }
 
-        private void ChangeTheme(Theme theme)
+        public void ChangeTheme(Theme theme, bool forceSystem)
         {
             Theme oldTheme = currentTheme;
 
-            if (theme == Theme.HighContrastOne)
+            if (theme == Theme.System)
+            {
+                currentTheme = Theme.System;
+                if (ControlzEx.Theming.WindowsThemeHelper.IsHighContrastEnabled())
+                {
+                    Theme highContrastBaseType = GetHighContrastBaseType();
+                    ChangeTheme(highContrastBaseType, true);
+                }
+                else
+                {
+                    string baseColor = ControlzEx.Theming.WindowsThemeHelper.GetWindowsBaseColor();
+                    ChangeTheme((Theme)Enum.Parse(typeof(Theme), baseColor), true);
+                }
+            }
+            else if (theme == Theme.HighContrastOne)
             {
-                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastOneTheme);
                 currentTheme = Theme.HighContrastOne;
+                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastOneTheme);
             }
             else if (theme == Theme.HighContrastTwo)
             {
-                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastTwoTheme);
                 currentTheme = Theme.HighContrastTwo;
+                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastTwoTheme);
             }
             else if (theme == Theme.HighContrastWhite)
             {
-                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastWhiteTheme);
                 currentTheme = Theme.HighContrastWhite;
+                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastWhiteTheme);
             }
             else if (theme == Theme.HighContrastBlack)
             {
-                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastBlackTheme);
                 currentTheme = Theme.HighContrastBlack;
+                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, HighContrastBlackTheme);
             }
             else if (theme == Theme.Light)
             {
-                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, LightTheme);
                 currentTheme = Theme.Light;
+                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, LightTheme);
             }
             else if (theme == Theme.Dark)
             {
-                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, DarkTheme);
                 currentTheme = Theme.Dark;
-            }
-            else
-            {
-                currentTheme = Theme.None;
+                ControlzEx.Theming.ThemeManager.Current.ChangeTheme(_app, DarkTheme);
             }
 
             ThemeChanged?.Invoke(oldTheme, currentTheme);
+
+            if (forceSystem)
```

Comment:
[clarification] Why is this required ? Wouldn't it be set when `theme` is Theme.System

Reply:
This is required when `theme == System.Theme` because `ChangeTheme` is called again passing the "real system color" but I need to keep `currentTheme = System.Theme`.
Otherwise if user change the system color again, the new color is not detected by the `ThemeManager`.
Required after removing direct access to settings.